test_runner/regress/test_timeline_archive.py

```diff
@@ -46,48 +48,46 @@ def test_timeline_archive(neon_simple_env: NeonEnv):
 
     assert exc.value.status_code == 404
 
-    # construct pair of branches to validate that pageserver prohibits
+    # construct a pair of branches to validate that pageserver prohibits
     # archival of ancestor timelines when they have non-archived child branches
-    parent_timeline_id = env.neon_cli.create_branch("test_ancestor_branch_archive_parent", "empty")
+    parent_timeline_id = env.neon_cli.create_branch("test_ancestor_branch_archive_parent")
 
     leaf_timeline_id = env.neon_cli.create_branch(
         "test_ancestor_branch_archive_branch1", "test_ancestor_branch_archive_parent"
     )
 
-    timeline_path = env.pageserver.timeline_dir(env.initial_tenant, parent_timeline_id)
-
     with pytest.raises(
         PageserverApiException,
         match="Cannot archive timeline which has non-archived child timelines",
     ) as exc:
-        assert timeline_path.exists()
-
         ps_http.timeline_archival_config(
             tenant_id=env.initial_tenant,
             timeline_id=parent_timeline_id,
             state=TimelineArchivalState.ARCHIVED,
         )
 
-    assert exc.value.status_code == 412
+    assert exc.value.status_code == 412 if unsharded else 409
 
     # Test timeline_detail
-    leaf_detail = ps_http.timeline_detail(
-        tenant_id=env.initial_tenant,
-        timeline_id=leaf_timeline_id,
-    )
-    assert leaf_detail["is_archived"] is False
+    if shard_count != 0:
```

Comment:
why is this only tested on sharded situation?

Reply:
`timeline_detail` is a pageserver-only API

Reply:
No it shouldn't be, I think it's transparently forwarded (the wildcard route). I may have queried it though directly on pageservers.

Reply:
okay good point, I've re-checked and it actually works. Removed the special casing.